### PR TITLE
source-woocommerce: ensure inline schemas, updated cdk, poetry, generate builder manifest

### PR DESCRIPTION
**Pull request by bleonard**
This was created from a set of automated scripts. In each case, not every update was needed for every connector, but overall here is what happened:

 * `auto-schema update -s all description`: added descriptions to json and inline schemas
 * `connector-code migrate_to_yaml -c all --type source`: migrates json schemas to connectors with a manifest
 * `airbyte-ci connectors --name=<all modified connectors from above> migrate_to_base_image`: makes sure that each is using a base docker image and updates docs
 * `airbyte-ci connectors --name=<all modified connectors from above> migrate-to-poetry`: moves connectors not already using poetry to do so and updates documentation
 * `airbyte-ci connectors --name=<all modified connectors from above> up_to_date`: updated the CDK to newer (0.88.0) version

The version number and changelogs were also bumped using the `connector-code pulls` command.

Additionally a few changes were made on top of the above changes:
* Generated manifest from builder, with embedded spec.
* Removed old spec and schema files.
* Added maxSecondsBetweenMessages to metadata.yaml
* Fixed a couple CATs failing because we no longer have JSON schema/spec.

### CODE REVIEW

File: docs/integrations/sources/woocommerce.md [hunk 1]
```diff
@@ -101,9 +101,13 @@ Useful links:
...
-| 0.2.3   | 2023-06-02 | [26955](https://github.com/airbytehq/airbyte/pull/26955) | Added `block_context` and `author` properties to the `Products` stream |
-| 0.2.2   | 2023-03-03 | [23599](https://github.com/airbytehq/airbyte/pull/23599) | Fix pagination and removed lookback window                             |
-| 0.2.1   | 2023-02-10 | [22821](https://github.com/airbytehq/airbyte/pull/22821) | Specified date formatting in specification                             |
-| 0.2.0   | 2022-11-30 | [19903](https://github.com/airbytehq/airbyte/pull/19903) | Migrate to low-code; Certification to Beta                             |
-| 0.1.1   | 2021-11-08 | [7499](https://github.com/airbytehq/airbyte/pull/7499)   | Remove base-python dependencies                                        |
-| 0.1.0   | 2021-09-09 | [5955](https://github.com/airbytehq/airbyte/pull/5955)   | Initial Release. Source WooCommerce                                    |
+| 0.2.7 | 2024-05-14 | [37289](https://github.com/airbytehq/airbyte/pull/37289) | Updating to 0.88.0 CDK |
```
Comment (lazebnyi) [inline docs/integrations/sources/woocommerce.md:104]: Have we published previous versions of the connector related to this PR?

File: airbyte-integrations/connectors/source-woocommerce/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+readme = "README.md"
+documentation = "https://docs.airbyte.com/integrations/sources/woocommerce"
+homepage = "https://airbyte.com"
+repository = "https://github.com/airbytehq/airbyte"
+[[tool.poetry.packages]]
+include = "source_woocommerce"
+
+[tool.poetry.dependencies]
+python = "^3.9,<3.12"
+airbyte-cdk = "0.88"
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-woocommerce/pyproject.toml:20]: Why do we hardcode from the minor version?

File: airbyte-integrations/connectors/source-woocommerce/README.md [hunk 3]
```diff
@@ -44,35 +63,41 @@ docker run --rm -v $(pwd)/secrets:/secrets airbyte/source-woocommerce:dev discov
...
 
-### Publishing a new version of the connector
+## Publishing a new version of the connector
 
 You've checked out the repo, implemented a million dollar feature, and you're ready to share your changes with the world. Now what?
 
 1. Make sure your changes are passing our test suite: `airbyte-ci connectors --name=source-woocommerce test`
-2. Bump the connector version in `metadata.yaml`: increment the `dockerImageTag` value. Please follow [semantic versioning for connectors](https://docs.airbyte.com/contributing-to-airbyte/resources/pull-requests-handbook/#semantic-versioning-for-connectors).
+2. Bump the connector version (please follow [semantic versioning for connectors](https://docs.airbyte.com/contributing-to-airbyte/resources/pull-requests-handbook/#semantic-versioning-for-connectors)): 
+    - bump the `dockerImageTag` value in in `metadata.yaml`
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-woocommerce/README.md:96]: Typo
```suggestion
    - bump the `dockerImageTag` value in `metadata.yaml`
```

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py [hunk 4]
```diff
@@ -717,7 +717,11 @@ def test_defined_cursors_exist_in_schema(self, discovered_catalog: Mapping[str,
         for stream_name, stream in discovered_catalog.items():
             if not stream.default_cursor_field:
                 continue
+
             schema = stream.json_schema
+            if not schema:
```
Comment (lazebnyi) [inline airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py:722]: It would be better to move those changes to another PR.

**Review by lazebnyi — CHANGES_REQUESTED** [review]

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py (hunk 4 above)
Comment (strosek) [reply airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py:722]: I added this change so the PR can run clean CATs, othrewise CI wouldn't pass. Is it OK to keep them?

File: airbyte-integrations/connectors/source-woocommerce/pyproject.toml (hunk 2 above)
Comment (strosek) [reply airbyte-integrations/connectors/source-woocommerce/pyproject.toml:20]: @maxi297 told me it was better to keep the connector stable in case of future changes.

File: docs/integrations/sources/woocommerce.md (hunk 1 above)
Comment (strosek) [reply docs/integrations/sources/woocommerce.md:104]: I don't think so. But the number changes came from the automated scripts run by Brian.

File: airbyte-integrations/connectors/source-woocommerce/pyproject.toml (hunk 2 above)
Comment (maxi297) [reply airbyte-integrations/connectors/source-woocommerce/pyproject.toml:20]: This was decided [here](https://airbytehq-team.slack.com/archives/C02TL38U5L7/p1713548724372399)

It seems like we are now reconsidering this [here](https://airbytehq-team.slack.com/archives/C02TL38U5L7/p1715906676083989)

**Comment by strosek** [comment]
Closed to avoid conflicts with Brian's work.
